ktor-client/ktor-client-core/common/src/io/ktor/client/features/observer/ResponseObserver.kt

```diff
@@ -42,7 +42,7 @@ class ResponseObserver(
         override fun install(feature: ResponseObserver, scope: HttpClient) {
 
             scope.receivePipeline.intercept(HttpReceivePipeline.Before) { response ->
-                val (loggingContent, responseContent) = response.content.split(response)
+                val (loggingContent, responseContent) = response.content.splitOnMainAndDependant(response)
```

Comment:
Could you tell me why we need new method here? It looks like the split should propagate cancellation to all channels

Reply:
Method `split` does split the channel and returns two fully asynchronous channels. As a result, logging can be fully done and completed even before the "main" channel will be canceled.

From my perspective, it's wrong and the logging channel should propagate only the data the main channel already read (it should wait for the main channel). To implement this idea we need a separate `splitOnMainAndDependant` method.